VirtoCommerce.Platform.Web/ApplicationInsights/CustomTelemetryInitializer.cs

```diff
@@ -0,0 +1,31 @@
+using System.Web;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace VirtoCommerce.Platform.Web.ApplicationInsights
+{
+    [System.CLSCompliant(false)]
+    public class CustomTelemetryInitializer : ITelemetryInitializer
+    {
+        public void Initialize(ITelemetry telemetry)
+        {
+            var httpContext = HttpContext.Current;
+
+            // Extend All telemetry with user identity context information.
+            if (httpContext != null && httpContext.User!=null)
+            {
+                 telemetry.Context.User.AuthenticatedUserId = httpContext.User.Identity.Name;
```

Comment:
Could you simplify this block and make the code more safe from null-referenced exceptions?
```C#
telemetry.Context.User.AuthenticatedUserId = httpContext?.User?.Identity?.Name;
```

Reply:
If `AuthenticatedUserId` was filled before, we will erase it in case of `null` context, user or identity. Though do not know if it is possible to have `AuthenticatedUserId` filled and have e.g. `null` identity.

Reply:
@tatarincev  I could not find any requirements for AuthenticatedUserId. 
Only that Authenticated user id should be a persistent string that uniquely represents each authenticated user in the application or service.